### PR TITLE
Core: Remove vite plugins and drop Vite 3 support

### LINKED ISSUE #25127: Support Vite 5
We want to add proper Vite 5 Support in Storybook 8.0.0. The problem is not about Vite 5, but maintaining and supporting multiple versions of Vite since we rely on Vite-plugins, which just support a particular version of Vite. Peer dependency issues are an issue that we have to overcome. One approach might be to make Vite not zero-configurable, but plugins have to be installed by the user to match the users Vite version. The following tasklist mentions the PRs, which are pending

```[tasklist]
- [x] Drop Vite < 4 support
- [ ] https://github.com/storybookjs/storybook/issues/25210
- [x] Update documentation
```

## Related PRs

```[tasklist]
- [ ] https://github.com/storybookjs/storybook/pull/25005
- [ ] https://github.com/storybookjs/storybook/pull/24953
- [ ] https://github.com/storybookjs/storybook/pull/24901
```

## Relevant issues

- https://github.com/storybookjs/storybook/issues/25125#issuecomment-1841761489
- https://github.com/storybookjs/storybook/issues/24918

### PR DESCRIPTION
**Pull request by kasperpeulen**
Relates to https://github.com/storybookjs/storybook/issues/25127
Closes https://github.com/storybookjs/storybook/pull/25427

## What I did

* Remove vite plugins
* Drop vite 3 support

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -336,6 +337,21 @@
 
 ## From version 7.x to 8.0.0
 
+### Framework specific vite plugins have to be explicitly added
```
Comment (valentinpalkovic) [inline MIGRATION.md:340]:
```suggestion
### Framework specific Vite plugins have to be explicitly added
```

File: MIGRATION.md [hunk 2]
```diff
@@ -336,6 +337,21 @@
...
+
+In Storybook 7 we would automatically add frameworks specific vite plugins, e.g. `@vitejs/plugin-react`, if they were not installed.
```
Comment (valentinpalkovic) [inline MIGRATION.md:342]:
```suggestion
In Storybook 7, we would automatically add frameworks-specific Vite plugins, e.g. `@vitejs/plugin-react` if not installed.
```

**Comment by valentinpalkovic** [comment]
@kasperpeulen Do you want us to test something explicitly?

**Comment by kasperpeulen** [comment]
@valentinpalkovic I don't think it is neccesary TBH. I did some manual testing on svelte and react, which was fine. I think problem with those changes should also be covered extensively in automated testing.

**Review by valentinpalkovic — APPROVED** [review]

File: MIGRATION.md [hunk 3]
```diff
@@ -336,6 +337,21 @@
 
 ## From version 7.x to 8.0.0
 
+### Framework-specific Vite plugins have to be explicitly added
```
Comment (yannbf) [inline MIGRATION.md:340]: Please add a note for Dropped Vite 3 support

File: code/frameworks/svelte-vite/package.json [hunk 4]
```diff
@@ -50,21 +50,22 @@
...
   },
   "devDependencies": {
+    "@sveltejs/vite-plugin-svelte": "^3.0.1",
     "@types/node": "^18.0.0",
     "svelte": "^5.0.0-next.16",
     "typescript": "^5.3.2",
     "vite": "^4.0.0"
   },
   "peerDependencies": {
+    "@sveltejs/vite-plugin-svelte": "^2.0.0 || ^3.0.0",
```
Comment (yannbf) [inline code/frameworks/svelte-vite/package.json:66]: You removed the plugin for react but added the one for svelte?

**Comment by IanVS** [comment]
Can I ask, why is it necessary to drop support for Vite 3?

**Pull request merged**